File: mindwatch-api/app/schemas/timeline.py
from pydantic import BaseModel
from datetime import datetime


class TimelineEntry(BaseModel):
    timestamp: datetime
    source: str
    severity: str
    risk_level: str
    confidence: float
    reasons: list[str]


class UserTimelineResponse(BaseModel):
    user_id: str
    timeline: list[TimelineEntry]

File: mindwatch-web/src/api/timeline.ts
import { api } from "./client";

export interface TimelineEntry {
    timestamp: string;
    source: string;
    severity: string;
    risk_level: string;
    confidence: number;
    reasons: string[];
}

export interface UserTimelineResponse {
    user_id: string;
    timeline: TimelineEntry[];
}

export async function fetchTimeline(): Promise<UserTimelineResponse> {
    const res = await api.get("/predict/timeline/me");
    return res.data;
}
